### PR TITLE
explorer: adjust blocks tables

### LINKED ISSUE #1030: Right-align numbers on blocks page
Please right-align all numbers on https://explorer.dcrdata.org/blocks , basically all columns except Time.

dcrdata v3.2.0-release

### PR DESCRIPTION
**Pull request by buck54321**
Makes adjustments to the tables on /blocks, /ticketpricewindows, and the time-based block groupings pages.

Adds block version column to /blocks (#1009).  /blocks has `explorertypes.BlockBasic`, so `Version` and `AllTxs` fields of `explorertypes.BlockInfo` graduated to fields in `BlockBasic`, which it already embeds. Appropriate changes were made in `makeExplorerBlockBasic`, including a change to remove an extraneous revocation counting loop. A `TxCount` field is added to `dbtypes.BlocksGroupedInfo` as well, for the grouped blocks listings.

JS: `humanize.formatTxDate` updated to follow up changes in #988, and renamed to `date`. `blocklist_controller` updated for new "version" field in websocket block.

Style changes to tables and pagination controls, mostly for mobile-friendliness. On smallest screen sizes, individual counts for 4 transaction types are hidden in favor of a single tx count column.

Also resolves #1030

### CODE REVIEW

File: explorer/explorerroutes.go [hunk 1]
```diff
@@ -477,6 +477,12 @@ func (exp *explorerUI) timeBasedBlocksListing(val string, w http.ResponseWriter,
 		return
 	}
 
+	// If the view is "years" and the top row is this year, modify the formatted
+	// time string to indicate its a partial result.
+	if val == "Years" && len(data) > 0 && data[0].EndTime.T.Year() == time.Now().Year() {
+		data[0].FormattedStartTime = fmt.Sprintf("%s so far", time.Now().Format("2006"))
```
Comment (buck54321) [inline explorer/explorerroutes.go]: Doing it this way prevents the need for a wide "end date" column on /years.

**Comment by chappjc** [comment]
#1077 is in now.

File: explorer/explorerroutes.go (hunk 1 above)
Comment (chappjc) [reply explorer/explorerroutes.go]: Makes a lot of sense.  I'm just wondering if YTD or "year to date" might be clearer than say "2019 so far".  I'm OK with this though.

Comment (buck54321) [reply explorer/explorerroutes.go]: Year to date is much better. Should it be hyphenated, "year-to-date" or not, "year to date"?

Comment (chappjc) [reply explorer/explorerroutes.go]: Good with YTD as you set it.  Let's see if we get comments when live.

**Pull request merged**